src/dataAccess/posts.ts

```diff
@@ -1,17 +1,30 @@
 import type { Post, PostResponse } from "@/types";
+import { BASE_API_URL } from "@/utils/constans";
```

Comment:
This could be inside of some HTTP client like axios or something custom (I'm slightly biased against axios 😆)
This would abstract away most of the endpoints and other repetitive stuff and focus on the application usage (with combination of some Api clients/services)